ens/utils.py

```diff
@@ -81,10 +81,12 @@ def normalize_name(name):
         return name
     elif isinstance(name, (bytes, bytearray)):
         name = name.decode('utf-8')
+    elif name[0:4] == 'xn--':
+        name = idna.decode(name, uts46=True, std3_rules=True)
     try:
-        return idna.decode(name, uts46=True, std3_rules=True)
+        return idna.uts46_remap(name, std3_rules=True)
```

Comment:
This appears to be where the failure about the leading `-` happened. I don't remember where that rule comes from, so it's worth making sure it's something actually required by ENS nameprep (and uts46).